x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/40_semantic_text_query.yml

```diff
@@ -90,6 +90,114 @@ setup:
   - close_to: { hits.hits.0._score: { value: 3.7837332e17, error: 1e10 } }
   - length: { hits.hits.0._source.inference_field.inference.chunks: 2 }
 
+---
+"Query using a sparse embedding model and inner chunks":
+  - requires:
+      cluster_features: "gte_v8.16.0"
+      reason: semantic_text inner hits support added in 8.16.0
+
+  - skip:
+      features: [ "headers", "close_to" ]
+
+  - do:
+      index:
+        index: test-sparse-index
+        id: doc_1
+        body:
+          inference_field: ["inference test", "another inference test"]
+          non_inference_field: "non inference test"
+        refresh: true
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
+        Content-Type: application/json
+      search:
+        index: test-sparse-index
+        body:
+          query:
+            semantic:
+              field: "inference_field"
+              query: "inference test"
+              inner_chunks: {}
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc_1" }
+  - close_to: { hits.hits.0._score: { value: 3.7837332e17, error: 1e10 } }
+  - length: { hits.hits.0._source.inference_field.inference.chunks: 2 }
+  - match: { hits.hits.0.inner_hits.chunks.hits.total.value: 2 }
+  - length: { hits.hits.0.inner_hits.chunks.hits.hits: 2 }
+  - match: { hits.hits.0.inner_hits.chunks.hits.hits.0._source.text: "another inference test" }
+  - match: { hits.hits.0.inner_hits.chunks.hits.hits.1._source.text: "inference test" }
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
+        Content-Type: application/json
+      search:
+        index: test-sparse-index
+        body:
+          query:
+            semantic:
+              field: "inference_field"
+              query: "inference test"
+              inner_chunks: {
+                "size": 1
```

Comment:
Can we also add tests for invalid from/size, e.g. -1? 
Could also be worth a test that specifies both size and from, adding a 3rd chunk to validate things got dropped off.